api/proto/teleport/accesslist/v1/accesslist.proto

```diff
@@ -84,6 +84,9 @@ message AccessListSpec {
 
   // owner_grants describes the access granted by owners to this access list.
   AccessListGrants owner_grants = 11;
+
+  // status contains dynamically calculated fields.
+  Status status = 12;
```

Comment:
Move this field outside of `spec` into the root `AccessList` object - similarly to what k8s does -  and name it `AccessListStatus`

The first is because the field isn't part of the access list spec and it only contains information and the second is because if we ever add other status to this package we will have weird namings.